File: CS/ReadOnlyCollectionWrapper/Form1.cs
using System;
using System.Windows.Forms;
using System.Collections.Generic;
using DevExpress.XtraGrid.Views.Base;

namespace ReadOnlyCollectionWrapper {
    public partial class Form1 :Form {
        List<string> descriptions;

        public Form1() {
            InitializeComponent();
            descriptions = new List<string>();
            for (int i = 0; i < source.Count; i++)
                descriptions.Add(string.Empty);
        }

        private void gridView1_CustomUnboundColumnData(object sender, CustomColumnDataEventArgs e) {
            if (descriptions == null) return;
            if (e.IsGetData) e.Value = descriptions[e.ListSourceRowIndex];
            else descriptions[e.ListSourceRowIndex] = (string)e.Value;
        }
    }
}

File: VB/ReadOnlyCollectionWrapper/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports System.Collections.Generic
Imports DevExpress.XtraGrid.Views.Base

Namespace ReadOnlyCollectionWrapper
	Partial Public Class Form1
		Inherits Form
		Private descriptions As List(Of String)

		Public Sub New()
			InitializeComponent()
			descriptions = New List(Of String)()
			For i As Integer = 0 To source.Count - 1
				descriptions.Add(String.Empty)
			Next i
		End Sub

		Private Sub gridView1_CustomUnboundColumnData(ByVal sender As Object, ByVal e As CustomColumnDataEventArgs) Handles gridView1.CustomUnboundColumnData
			If descriptions Is Nothing Then
				Return
			End If
			If e.IsGetData Then
				e.Value = descriptions(e.ListSourceRowIndex)
			Else
				descriptions(e.ListSourceRowIndex) = CStr(e.Value)
			End If
		End Sub
	End Class
End Namespace